WhateverGreen/kern_igfx.cpp

```diff
@@ -141,7 +141,12 @@ void IGFX::init() {
 }
 
 void IGFX::deinit() {
-
+	for (int index = 0; index < arrsize(lspcons); index++) {
+		if (lspcons[index].lspcon != nullptr) {
+			delete lspcons[index].lspcon;
```

Comment:
Please do not use new/delete directly in XNU. Under normal conditions delete must check for nullptr according to C++ standard, but Apple's implementation does not, and thus will crash. See: https://github.com/apple/darwin-xnu/blob/a449c6a3b8014d9406c2ddbdc81795da24aa7443/libkern/c%2B%2B/OSRuntime.cpp#L543

It is very confusing for many people, and thus we invented a concept of `create` and `deleter` methods. Probably it is a good idea to implement them here too, like this:
https://github.com/acidanthera/Lilu/blob/223f20f030d575f9782880e3a7e520fe4dcfc1cd/Lilu/Headers/kern_mach.hpp#L176-L177

Reply:
```
if (lspcons[index].lspcon != nullptr) {
	delete lspcons[index].lspcon;
}
```
I think I have checked for nullptr before I delete the instance...  

Anyway, I got your idea and have implemented the `creater` and `deleter` methods as suggested, but I have a question related to the implementation of `MachInfo::deleter()`.   Shouldn't we check whether the instance is nullptr or not before deleting it?  Otherwise, what's the point of adding a deleter method that just `delete` the given instance?

```
		static void deleter(LSPCON *instance) {
			if (instance != nullptr)
				delete instance;
		}
```

Reply:
No, it should definitely be caller-dependent. The original logic was obscure due to no non-null attribute and sometimes inconsistent code. I fixed this in https://github.com/acidanthera/Lilu/commit/f45255eb785a2a2728bb07815e50cb8219bfeb31.

Reply:
Got it. :D